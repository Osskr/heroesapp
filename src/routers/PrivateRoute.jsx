import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import {PropTypes} from 'prop-types'
export const PrivateRoute = ({isAuthenticated,component: Component, ...rest}) => {
    
    
    console.log(rest.location.pathname )
    //guardamos la ruta en el local storage
    localStorage.setItem('lastPath',rest.location.pathname)
    return (
        <Route {...rest}  
               component={ (props)=>(
                          (isAuthenticated)? (<Component {...props}/>)
                          : (<Redirect to='login'/>)
                )}/>

     
    )
}
 
PrivateRoute.propTypes ={

    isAuthenticated : PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired,
       
}

