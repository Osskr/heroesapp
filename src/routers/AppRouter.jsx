import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch,
  }
from "react-router-dom";
import { AuthContext } from "../auth/AuthContex";

import { LoginScreen } from "../components/login/LoginScreen";
import { DashBoardRoutes } from "./DashBoardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {

 const {user} = useContext(AuthContext)
  
    return (
        <div>
            <Router>
                <div>                                    
                    <Switch>
                        <PublicRoute  isAuthenticated={user.logged} exact path="/login" component={LoginScreen}/>
                        <PrivateRoute path="/"
                                      component={DashBoardRoutes}
                                      isAuthenticated={user.logged}/>
                    </Switch>                
                </div>
            </Router>
        </div>
    )
}
