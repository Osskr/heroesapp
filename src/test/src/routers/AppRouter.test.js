import React from 'react'
import {mount} from 'enzyme'
import { AppRouter } from '../../../routers/AppRouter'
import { AuthContext } from '../../../auth/AuthContex'




describe('Pruebas sobre el AppRouter', () => {
  
  
  const contextValue = {
    dispatch:jest.fn(),
    user:{
      logged:false
    }
  }
  test('debe mostrar el login si no estoy autenticado', () => {

      const wrapper = mount(
      
          <AuthContext.Provider value={contextValue}>
              <AppRouter/>
          </AuthContext.Provider>)
      
      expect(wrapper).toMatchSnapshot()
  })
  

  test('debe mostrar el componente de marvel si esta autenticado',()=>{
      // voy a suponer que estoy autenticado
      const contextValue = {
        dispatch:jest.fn(),
        user:{
          logged:true,
          name:'sskr'
        }
      }
    const wrapper = mount(
      
      <AuthContext.Provider value={contextValue}>
          <AppRouter/>
      </AuthContext.Provider>)

    expect(wrapper.find('.navbar').exists()).toBe(true)
  })
})
