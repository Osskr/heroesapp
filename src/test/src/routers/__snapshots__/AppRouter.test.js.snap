// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el AppRouter debe mostrar el login si no estoy autenticado 1`] = `
<div>
  <div>
    <div
      className="container mt-5"
    >
      <h1>
        Login Screen
      </h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
